telegram/ext/conversationhandler.py

```diff
@@ -354,12 +354,12 @@ def update_state(self, new_state, key):
             if self.persistent:
                 self.persistence.update_conversation(self.name, key, new_state)
 
-    def _trigger_timeout(self, dispatcher):
+    def _trigger_timeout(self, context):
         self.logger.debug('conversation timeout was triggered!')
-        del self.timeout_jobs[dispatcher.job.context.conversation_key]
+        del self.timeout_jobs[context.job.context.conversation_key]
         handlers = self.states.get(self.TIMEOUT, [])
         for handler in handlers:
-            check = handler.check_update(dispatcher.job.context.update)
+            check = handler.check_update(context.job.context.update)
             if check is not None and check is not False:
-                handler.handle_update(dispatcher.job.context.update, dispatcher.job.context.dispatcher, check)
-        self.update_state(self.END, dispatcher.job.context.conversation_key)
+                handler.handle_update(context.job.context.update, context.job.context.dispatcher, check, context=context)
```

Comment:
@jsmnbom I've made the changes but I want to double check if:

`context.job.context.update` --> `context.update`
`context.job.context.dispatcher` --> `context.dispatcher`

```
>>> dir(context)
['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_dispatcher', 'args', 'bot', 'chat_data', 'error', 'from_error', 'from_job', 'from_update', 'job', 'job_queue', 'match', 'matches', 'update', 'update_queue', 'user_data']
```

Reply:
Oh no I'm wrong here. It looks like context doesn't actually store the dispatcher, but the job's context does have a dispatcher so context.job.context.update and context.job.context.dispatcher are the correct ones to use.

The reason you see a context.update is that it's a method used to update the context dict (that and how the context gets constructed can be found here https://github.com/python-telegram-bot/python-telegram-bot/blob/master/telegram/ext/callbackcontext.py#L103)

Reply:
Ohhh that explains. Alright I'll just leave it for now.